app/models/host/discovered.rb

```diff
@@ -126,7 +132,12 @@ def populate_fields_from_facts(parser, type, source_proxy)
   def populate_discovery_fields_from_facts(facts)
     ForemanDiscovery::ImportHookService.new(host: self, facts: facts).after_populate
   ensure
-    self.save!
+    begin
+      self.save!
+    rescue => _
+      telemetry_increment_counter(:failed_discovered_hosts)
+      raise
```

Comment:
Would you want to raise an error or default to `StandardError`?

Reply:
Calling raise without any arguments will raise the last error. In our case, we are re-raising error. This is a standard idiom in many programming languages, I have this from Java and I believe Ruby does the same.